batch.go

```diff
@@ -0,0 +1,126 @@
+package ed25519consensus
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/sha512"
+
+	"filippo.io/edwards25519"
+)
+
+// Verifier holds entries of public keys, signature and a scalar which are used for batch verification.
+type Verifier struct {
+	entries   []ks
+	batchSize uint32
+}
+
+// ks represents the public key, signature and scalar which the caller wants to batch verify
+type ks struct {
+	pubkey    ed25519.PublicKey
+	signature []byte
+	k         *edwards25519.Scalar
+}
+
+// NewVerifier creates a Verifier that entries of signatures, keys and messages
+// can be added to for verification
+func NewVerifier() Verifier {
+	return Verifier{
+		entries:   []ks{},
+		batchSize: 0,
+	}
+}
+
+// Add adds an entry to the verifier and bumps the batch size.
+func (v *Verifier) Add(publicKey ed25519.PublicKey, sig, message []byte) bool {
+	if l := len(publicKey); l != ed25519.PublicKeySize {
+		return false
+	}
+
+	if len(sig) != ed25519.SignatureSize || sig[63]&224 != 0 {
+		return false
+	}
+
+	h := sha512.New()
+	h.Write(sig[:32])
+	h.Write(publicKey)
+	h.Write(message)
+	var digest [64]byte
+	h.Sum(digest[:0])
+
+	k := new(edwards25519.Scalar).SetUniformBytes(digest[:])
+
+	ksS := ks{
+		pubkey:    publicKey,
+		signature: sig,
+		k:         k,
+	}
+
+	v.entries = append(v.entries, ksS)
+	v.batchSize++
+
+	return true
+}
+
+// VerifyBatch batch verifies the keys, messages and signatures within the verifier.
+// If a failure arises it is unknown which key failed, the caller must verify each entry individually
+func (v *Verifier) VerifyBatch() bool {
```

Comment:
If `Verifier` becomes `BatchVerifier` as in https://github.com/hdevalence/ed25519consensus/pull/7/files#r564128054, then `VerifyBatch` can be shortened to just `Verify`:
```suggestion
// Verify checks all entries in the current batch, returning `true` if 
// *all* entries are valid and `false` if *any one* entry is invalid.
//
// If a failure arises it is unknown which entry failed, the caller must verify each entry individually.
func (v *BatchVerifier) Verify() bool {
```